TaskSharper.Calender.WPF/Views/Components/DateTimePickerView.xaml.cs

```diff
@@ -0,0 +1,198 @@
+            }
+        }
+
+        public object Date
+        {
+            get => GetValue(DateProperty);
+            set => SetValue(DateProperty, value);
+        }
+
+        public static readonly DependencyProperty DateProperty =
+            DependencyProperty.Register("Date", typeof(object), typeof(DateTimePickerView), new PropertyMetadata(null));
+        
+        private void SelectedDatesChanged(object sender, SelectionChangedEventArgs e)
+        {
+            var calendar = (Calendar) sender;
+            var selectedDate = calendar.SelectedDate.Value;
+            
+            Date = new DateTime(selectedDate.Year, selectedDate.Month, selectedDate.Day) + ((DateTime)Date).TimeOfDay;
+        }
+
+        private void CalendarLoaded(object sender, RoutedEventArgs e)
+        {
+            DatePickerCalendar.SelectedDate = (DateTime) Date;
+            DatePickerCalendar.DisplayDate = ((DateTime) Date).Date;
+
+            HourListBox.SelectedIndex = ((DateTime)Date).Hour;
+            MinuteListBox.SelectedIndex = ((DateTime) Date).Minute;
+
+            HourListBox.ScrollToCenterOfView(HourListBox.Items[HourListBox.SelectedIndex]);
+            MinuteListBox.ScrollToCenterOfView(MinuteListBox.Items[MinuteListBox.SelectedIndex]);
+        }
+
+        private void HourChanged(object sender, SelectionChangedEventArgs e)
+        {
+            var selectedHour = (int) ((ListBox) sender).SelectedItem;
+            var currentDate = (DateTime) Date;
+
+            Date = new DateTime(currentDate.Year, currentDate.Month, currentDate.Day, selectedHour, currentDate.Minute, currentDate.Second);
+            HourListBox.ScrollToCenterOfView(HourListBox.SelectedItem);
+        }
+
+        private void MinuteChanged(object sender, SelectionChangedEventArgs e)
+        {
+            var selectedMinute = (int)((ListBox)sender).SelectedItem;
+            var currentDate = (DateTime)Date;
+
+            Date = new DateTime(currentDate.Year, currentDate.Month, currentDate.Day, currentDate.Hour, selectedMinute, currentDate.Second);
+            MinuteListBox.ScrollToCenterOfView(MinuteListBox.SelectedItem);
+        }
+
+        private void HourScrollUp(object sender, MouseButtonEventArgs e)
+        {
+            HourListBox.DecrementIndex(0, 23, 1);
+        }
+
+        private void HourScrollDown(object sender, MouseButtonEventArgs e)
+        {
+            HourListBox.IncrementIndex(0, 23, 1);
+        }
+
+        private void MinuteScrollUp(object sender, MouseButtonEventArgs e)
+        {
+            MinuteListBox.DecrementIndex(0, 59, 5);
+        }
+
+        private void MinuteScrollDown(object sender, MouseButtonEventArgs e)
+        {
+            MinuteListBox.IncrementIndex(0, 59, 5);
+        }
+
+        void ScrollViewer_ManipulationBoundaryFeedback(object sender, ManipulationBoundaryFeedbackEventArgs e)
+        {
+            e.Handled = true;
+        }
+    }
+
+    
+    public static class ItemsControlExtensions
```

Comment:
Syntes stadig vi skal sepererer extension methods ud for sig selv - men det kan vente :)